plugin.video.arloview/resources/lib/arlo_stream.py

```diff
@@ -0,0 +1,275 @@
+        self.debug_mode = REAL_SETTINGS.getSetting('enable_debug') == 'true'
+        self.log("END   __init__", xbmc.LOGDEBUG)
+
+    def log(self, msg, level=xbmc.LOGNOTICE):
+        if level < self.log_level:
+            return
+        # if level == xbmc.LOGERROR:
+        #    msg += ' ,' + traceback.format_exc()
+        if level < xbmc.LOGNOTICE:
+            level = xbmc.LOGNOTICE
+        xbmc.log("[{}-{}] {}".format(ADDON_ID, ADDON_VERSION, msg), level)
+
+    def check_first_run(self):
+
+        if REAL_SETTINGS.getSetting("userid") == "":
+            self.log("check_first_run()- userid NOT set!", xbmc.LOGNOTICE)
+            msg = "Set ARLO credentials, otherwise...\nVideo will not stream correctly!"
+            icon = REAL_SETTINGS.getAddonInfo('icon')
+            xbmc.executebuiltin('Notification(%s, %s, %d, %s)' % (ADDON_NAME, msg, 5000, icon))
+            REAL_SETTINGS.openSettings()
+            self.debug_mode = REAL_SETTINGS.getSetting('enable_debug') == 'true'
+            self.log_level = int(REAL_SETTINGS.getSetting("log_level"))
+
+    def main_menu(self):
+        self.log("BEGIN main_menu()", xbmc.LOGNOTICE)
+
+        for camera in self._get_arlo_cameras():
+            camera_info = self._get_camera_info(camera)
+            list_item = xbmcgui.ListItem(label=camera["deviceName"])
+            list_item.setInfo('video', {'title': camera["deviceName"],
+                                        'plot': camera_info,
+                                        'mediatype': 'video'})
+            list_item.setProperty('IsPlayable', 'true')
+            snapshot_file = self._get_camera_snapshot(camera['deviceId'])
+            list_item.setArt({'thumb': snapshot_file,
+                              'icon': snapshot_file,
+                              'poster': snapshot_file,
+                              'clearart': snapshot_file,
+                              'clearlogo': snapshot_file})
+            # Create a URL for a plugin recursive call.
+            url = get_url("plugin://" + ADDON_ID,
+                          cameraName=camera["deviceName"],
+                          cameraId=camera["deviceId"])
+            # Add the list item to a virtual Kodi folder.
+            # is_folder = False means that this item won't open any sub-list.
+            is_folder = False
+            # Add our item to the Kodi virtual folder listing.
+            xbmcplugin.addDirectoryItem(self._handle, url, list_item, is_folder)
+        # Add a sort method for the virtual folder items (alphabetically, ignore articles)
+        xbmcplugin.addSortMethod(self._handle, xbmcplugin.SORT_METHOD_LABEL_IGNORE_THE)
+        # Finish creating a virtual folder.
+        xbmcplugin.endOfDirectory(self._handle)
+
+        self.log("  Setup refresh timer", xbmc.LOGDEBUG)
+        # Update the gui to reflect signal and battery strength
+        gui_update_timer = threading.Timer(120, self._refresh_gui)
+        gui_update_timer.start()
```

Comment:
This will block the python script execution from finishing for 2 minutes. That's not allowed.